system/Entity.php

```diff
@@ -388,6 +396,22 @@ protected function castAs($value, string $type)
 					$value = unserialize($value);
 				}
 
+				$value = (array)$value;
+				break;
+			case 'json':
+				if (function_exists('json_decode') && is_string($value) && (strpos($value, '[') === 0 || strpos($value, '{') === 0))
+				{
+					$value = json_decode($value, false);
```

Comment:
You should check the json_error_code here, also, to ensure it's valid json, or throw an exception otherwise, so the developer knows what went wrong.

Reply:
Firstly I am not convinced about throwing exception when case 'array'  won't throw anything when serialized array is formatted  incorrectly. 

Secondly if you think it is really required to throw exception with json then we have to clarify that
json_error_code() will return error if $value is null; what could be very likely case when data is retrieved from database. Do you reckon we should treat NULL value as exception or just ignore it? i think i should ignore it in that case, as we do in the moment.

--
i will add docs and test in tmrw.

Reply:
exception + language  (messages) files added and docs modified